File: gems/decomplex/examples/rust/state-mesh.rs
struct StateMeshExample {
    a: i32,
    b: i32,
}

impl StateMeshExample {
    fn initialize(&mut self) {
        self.a = 1;
        self.b = 2;
    }

    fn writer(&mut self) {
        self.a = 3;
    }

    fn reader(&self) -> i32 {
        self.a + self.b
    }

    fn a_alias(&self) -> i32 {
        self.a
    }
}

File: gems/decomplex/examples/zig/state-mesh.zig
const StateMeshExample = struct {
    a: i32,
    b: i32,

    pub fn initialize(self: *StateMeshExample) void {
        self.a = 1;
        self.b = 2;
    }

    pub fn writer(self: *StateMeshExample) void {
        self.a = 3;
    }

    pub fn reader(self: *StateMeshExample) i32 {
        return self.a + self.b;
    }

    pub fn a_alias(self: *StateMeshExample) i32 {
        return self.a;
    }
};
